tensorflow_io/frame/python/ops/frame_ops.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""DataFrame"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+import tensorflow as tf
+from tensorflow_io.text.python.ops import text_ops
+from tensorflow_io.core.python.ops import data_ops
+
+class _ILocIndexer(object):
+  """_ILocIndexer"""
+  def __init__(self, df):
+    self._df = df
+    return
+
+  def __getitem__(self, item):
+    data = dict(
+        (column, tf.Variable(self._df._data[column][item]) # pylint: disable=protected-access
+        ) for column in self._df.columns)
+    return DataFrame(data, columns=list(self._df.columns))
+
+
+class DataFrame(object):
+  """DataFrame
+
+  NOTE: Experimental and eager only!
+  """
+
+  def __init__(self, data=None, columns=None):
+    """Create a DataFrame."""
+    if not tf.executing_eagerly():
+      raise NotImplementedError("DataFrame only supports eager mode")
+    self._data = data
+    self._columns = columns
+    self._index = np.arange(data[columns[0]].shape[0].value)
+    self._iloc = _ILocIndexer(self)
+
+  @property
+  def columns(self):
+    return self._columns
+
+  @property
+  def index(self):
+    return self._index
+
+  @property
+  def values(self):
+    """values"""
+    if len(self._columns) == 1:
+      return self._data[self._columns[0]].numpy()
+    l = [tf.int32, tf.int64, tf.float32, tf.float64, tf.string]
+
+    dtype = l[
+        max([l.index(self._data[column].dtype) for column in self.columns])]
+    if dtype == tf.string:
+      raise NotImplementedError("DataFrame does not support to np.object")
+    raise NotImplementedError("DataFrame does not support values yet")
+
+  @property
+  def shape(self):
+    return len(self.index), len(self.columns)
+
+  @property
+  def iloc(self):
+    return self._iloc
+
+  def __len__(self):
+    return len(self._index)
+
+  def __iter__(self):
+    return iter(self._index)
```

Comment:
I think you would want to iterate over `_data` not the index right?

Reply:
Yes. Updated